lib/api/BackbeatRequest.js

```diff
@@ -0,0 +1,137 @@
+'use strict'; // eslint-disable-line
+
+/**
+ * Class representing a Backbeat API Request
+ *
+ * @class
+ */
+class BackbeatRequest {
+    constructor() {
+        this._log = null;
+        this._request = null;
+        this._response = null;
+        this._statusCode = 0;
+        this._route = null;
+        this._error = null;
+    }
+
+    /**
+     * Get logger object
+     * @return {object} Logger object
+     */
+    getLog() {
+        return this._log;
+    }
+
+    /**
+     * Set logger object
+     * @param {object} log - new Logger object
+     * @return {BackbeatRequest} itself
+     */
+    setLog(log) {
+        this._log = log;
+        return this;
+    }
+
+    /**
+     * Get http request object
+     * @return {object} Http request object
+     */
+    getRequest() {
+        return this._request;
+    }
+
+    /**
+     * Set http request object
+     * @param {object} request - new Http request object
+     * @return {BackbeatRequest} itself
+     */
+    setRequest(request) {
+        this._request = request;
+        return this;
+    }
+
+    /**
+     * Get http response object
+     * @return {object} Http response object
+     */
+    getResponse() {
+        return this._response;
+    }
+
+    /**
+     * Set http response object
+     * @param {object} response - new Http response object
+     * @return {BackbeatRequest} itself
+     */
+    setResponse(response) {
+        this._response = response;
+        return this;
+    }
+
+    /**
+     * Get status code of request
+     * @return {number} Http status code
+     */
+    getStatusCode() {
+        return this._statusCode;
+    }
+
+    /**
+     * Set status code of request
+     * @param {number} code - new Http status code
+     * @return {BackbeatRequest} itself
+     */
+    setStatusCode(code) {
+        this._statusCode = code;
+        return this;
+    }
+
+    /**
+     * Get route
+     * @return {string} current route
+     */
+    getRoute() {
+        return this._route;
+    }
+
+    /**
+     * Set route
+     * @param {string} route - new route string
+     * @return {BackbeatRequest} itself
+     */
+    setRoute(route) {
+        this._route = route;
+        return this;
+    }
+
+    /**
+     * Status check to see if valid request
+     * @return {boolean} valid status check
+     */
+    getStatus() {
+        return ((this.getStatusCode >= 200 && this.getStatusCode < 300)
+            && Object.keys(this.getError()).length === 0);
```

Comment:
You should check what is error type first before considering browsing the keys, because it may be `null`. Why not just add a `!this._error` check instead?